### ISSUE TITLE
HELICS values sent as string not being cleared?

### ISSUE THREAD

**Issue by trevorhardy** [gridlab-d/gridlab-d#1285]
- [x] I have not found an open issue describing the same problem.
- [x] I went through "CONTRIBUTING.MD" before deciding to submit the issue.
- [x] While building GridLAB-D, I followed all steps as detailed in GridLAB-D Wiki.

GLD version: 1279b
HELICS version: 3.0.0alpha2
Python version: 3.8.5
------------------------------------------------------------------------------------------------------------

## Please describe what was the cause of the issue?
Sending multiple value signals via HELICS as strings are somehow getting messed up.

## Please provide the steps to reproduce.
See attached test case
Switch status are being sent as string value signals are not being properly cleared from a common buffer (maybe?) and are getting mangled together. 

Two switches in GridLAB-D are being set via a Python federate. Non-deterministically (due to HELICS?) an error message shows up in GridLAB-D indicating that the status property was not valid.

## Please provide the exact error/warnining/failure message printed on the console.
When `transformerSwitchBad` is set to `OPEN` in GridLAB-D the error message shows up as:
```
keyword 'OPENED' is not valid for property status
```

When `transformerSwitchBad` is set to `OPENS` in GridLAB-D the error message shows up as:
```
keyword 'OPENSD' is not valid for property status
```

## If possible, please provide any screenshots and other supporting information.
[bug_case.zip](https://github.com/gridlab-d/gridlab-d/files/5896559/bug_case.zip)

-------------------------------------------------------------------------------------------------------------

**Comment by blthayer** [comment]
@trevorhardy @afisher1 @eberleim @ftuffner 

I am also encountering a similar issue when commanding capacitors. Relevant logging snippets from my program:

```
...
command_federate: Sending capacitor state OPEN to tap_controller/cap_state_to_gld.
...
gridlab-d: ERROR    [2018-01-01 00:00:00 UTC] : keyword 'OPENSV' is not valid for property switchA
```

**Comment by laurmarinovici** [comment]
@trevorhardy @afisher1 @eberleim @ftuffner @blthayer 

I got similar behavior when commanding switches. Co-sim configuration:
- HELICS - 2.7.0 (2021-04-29)
- GridLAB-D - 4.3.0-18718 (Navajo [57eb0dc4:feature/730]) 64-bit LINUX RELEASE
- Python - 3.9.0+

```
...
Python federate command: switch GLD/sub_sw13to152 to OPEN
...
gridlab-d: ERROR    [2001-08-01 13:00:01 PDT] : keyword 'OPENOV' is not valid for property status
```

**Comment by trevorhardy** [comment]
@nathantgray discovered that null terminating the string (_e.g._ `OPEN\0`) tricks the GridLAB-D parser into handling the string appropriately. Hopefully this bandaid will work while the bug itself gets fixed.

**Comment by laurmarinovici** [comment]
Could this be caused in conjunction with something else?

I just ran my code again on the same Docker environment, just couple of days later, and it worked. See below the results from the GLD recorder, particularly after 3 seconds for sw60to160 and 4 seconds for tie_sw_13_95.

| # timestamp | sw60to160 | tie_swt_13_95
| --- | --- | --- |
2001-08-01 13:00:00 PDT | CLOSED | OPEN
2001-08-01 13:00:01 PDT | CLOSED | OPEN
2001-08-01 13:00:02 PDT | CLOSED | OPEN
2001-08-01 13:00:03 PDT | OPEN | OPEN
2001-08-01 13:00:04 PDT | OPEN | CLOSED
2001-08-01 13:00:05 PDT | OPEN | CLOSED
2001-08-01 13:00:06 PDT | OPEN | CLOSED
2001-08-01 13:00:07 PDT | OPEN | CLOSED
2001-08-01 13:00:08 PDT | OPEN | CLOSED

The Python federate code that is sending the switching commands is:
```
   if len(commandOpen) > 0:
      stats = 'OPEN'
      for cOpen in commandOpen:
        destination = f'GLD/disp_{cOpen}'
        messg = stats
        logger.info('********** At {0} switch {1} to {2} **********'.format(currTime, destination, messg))
        h.helicsEndpointSendMessageRaw(fedEP, destination, messg)
    if len(commandClose) > 0:
      stats = 'CLOSED'
      for cClose in commandClose:
        destination = f'GLD/disp_{cClose}'
        messg = stats
        logger.info('********** At {0} switch {1} to {2} **********'.format(currTime + deltaTime, destination, messg))
        # This command will delay the message sending.
        h.helicsEndpointSendEventRaw(fedEP, destination, messg, currTime + deltaTime)
```

**Comment by eberleim** [comment]
Fixed by PR #1304 and merged to develop.
Branches that still exhibit this bug should bring in the fix from develop.
